gapis/api/vulkan/api/renderpass_framebuffer.api

```diff
@@ -224,13 +224,155 @@ Thefore there is no need to repeat the KHR structs
   @unused RenderPassVersion                  Version
 }
 
-sub AttachmentReference convertAttachment(VkAttachmentReference attachment) {
-  return AttachmentReference(
-    Attachment: attachment.attachment,
-    Layout: attachment.layout,
-    AspectMask: 0, // This will be updated by VkRenderPassInputAttachmentAspectCreateInfo
+sub AttachmentDescription createAttachmentDescriptionObject(VkAttachmentDescription attachmentDescription) {
+ internalAttachmentDescription := AttachmentDescription(
+    Flags: attachmentDescription.flags,
+    Format: attachmentDescription.format,
+    Samples: attachmentDescription.samples,
+    LoadOp: attachmentDescription.loadOp,
+    StoreOp: attachmentDescription.storeOp,
+    StencilLoadOp: attachmentDescription.stencilLoadOp,
+    StencilStoreOp: attachmentDescription.stencilStoreOp,
+    InitialLayout: attachmentDescription.initialLayout,
+    FinalLayout: attachmentDescription.finalLayout,
     StencilLayout: null,
   )
+  return internalAttachmentDescription
+}
+
+sub AttachmentDescription createAttachmentDescription2Object(VkAttachmentDescription2 attachmentDescription2) {
+  internalAttachmentDescription := AttachmentDescription(
+    Flags: attachmentDescription2.flags,
+    Format: attachmentDescription2.format,
+    Samples: attachmentDescription2.samples,
+    LoadOp: attachmentDescription2.loadOp,
+    StoreOp: attachmentDescription2.storeOp,
+    StencilLoadOp: attachmentDescription2.stencilLoadOp,
+    StencilStoreOp: attachmentDescription2.stencilStoreOp,
+    InitialLayout: attachmentDescription2.initialLayout,
+    FinalLayout: attachmentDescription2.finalLayout,
+    StencilLayout: null,
+  )
+  if attachmentDescription2.pNext != null {
+    numPNext := numberOfPNext(attachmentDescription2.pNext)
+    next := MutableVoidPtr(as!void*(attachmentDescription2.pNext))
+    for j in (0 .. numPNext) {
+      sType := as!const VkStructureType*(next.Ptr)[0]
+      switch sType {
+        case VK_STRUCTURE_TYPE_ATTACHMENT_DESCRIPTION_STENCIL_LAYOUT: {
+          ext := as!VkAttachmentDescriptionStencilLayout*(next.Ptr)[0]
+          internalAttachmentDescription.StencilLayout = new!AttachmentDescriptionStencilLayout(
+            StencilInitialLayout: ext.stencilInitialLayout,
+            StencilFinalLayout: ext.stencilFinalLayout,
+          )
+        }
+      }
+    }
+  }
+  return internalAttachmentDescription
+}
+
+sub SubpassDescription createSubpassDescriptionObject(VkSubpassDescription2 subpassDescription2) {
```

Comment:
Maybe we can also have the function that creates `SubpassDescription` from `VkSubpassDescription` as well. Similar to `VkAttachmentDescription` and `VkAttachmentDescription2`?

Reply:
Yes, Done.

Reply:
Thanks!